### PR TITLE
Export BuildArgs type from dockerfile package

### PR DESCRIPTION
**Pull request by priyawadhwa**
#37015

**- What I did**
I exported BuildArgs so that I could use it in [kaniko](https://github.com/GoogleContainerTools/kaniko) to help support the ARG command.

**- How I did it**
n/a

**- How to verify it**
n/a

**- Description for the changelog**
Export BuildArgs type.

### CODE REVIEW

File: builder/dockerfile/buildargs.go [hunk 1]
```diff
@@ -33,16 +33,17 @@ type buildArgs struct {
 	argsFromOptions map[string]*string
 }
 
-func newBuildArgs(argsFromOptions map[string]*string) *buildArgs {
-	return &buildArgs{
+func newBuildArgs(argsFromOptions map[string]*string) *BuildArgs {
```
Comment (AkihiroSuda) [inline builder/dockerfile/buildargs.go]: Can we expose this as well?

Comment (priyawadhwa) [reply builder/dockerfile/buildargs.go]: for sure, done!

**Comment by tonistiigi** [comment]
@priyawadhwa lint/build error in ci

**Comment by priyawadhwa** [comment]
@tonistiigi thanks, I fixed the errors. I think we might need to rerun the experimental tests (looks like one of the tests timed out)

**Comment by tonistiigi** [comment]
LGTM but squash the commits please

**Comment by priyawadhwa** [comment]
Done! I think we just need to rerun the failing tests, looks like it timed out again (they were both passing before and I didn't change anything)

**Review by boaz0 — APPROVED** [review]
LGTM :+1: if all green.

**Comment by ripcurld** [comment]
janky's and powerpc's failures are unrelated to the change.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**